osquery/tables/utility/file.cpp

```diff
@@ -99,6 +127,171 @@ void genFileInfo(const fs::path& path,
+                              NULL,
+                              &pSD);
+
+  // Check GetLastError for GetSecurityInfo error condition.
+  if (dwRtnCode != ERROR_SUCCESS) {
+    return;
+  }
+
+  FILE_BASIC_INFO basicInfo;
+  BY_HANDLE_FILE_INFORMATION fileInfo;
+
+  if (0 == GetFileInformationByHandle(hFile, &fileInfo)) {
+    return;
+  }
+
+  auto fileIndex = ((unsigned long long)fileInfo.nFileIndexHigh << 32) +
+                   (unsigned long long)fileInfo.nFileIndexLow;
+
+  std::stringstream stream;
+  stream << "0x" << std::setfill('0')
+         << std::setw(sizeof(unsigned long long) * 2) << std::hex << fileIndex;
+  std::string fileId(stream.str());
+
+  // Windows has fileid's that are displayed in hex using: 
+  // fsutil file queryfileid <filename>
+  r["fileId"] = TEXT(fileId);
+
+  // inode is the decimal equivalent of fileid
+  r["inode"] = BIGINT(fileIndex);
+  r["uid"] = INTEGER(getUidFromSid(pSidOwner));
+  r["gid"] = INTEGER(getGidFromSid(pSidOwner));
+
+  // Permission bits don't make sense for Windows. Use ntfs_acl_permissions
+  // table
+  r["mode"] = "-1";
+
+  auto dwfileType = GetFileType(hFile);
+
+  // Try to assign a human readable file type
+  switch (dwfileType) {
+  case FILE_TYPE_CHAR:
+
+    r["type"] = "Character";
+    break;
+
+  case FILE_TYPE_DISK:
+
+    if (fileInfo.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) {
+      r["type"] = "Directory";
+
+    } else if (fileInfo.dwFileAttributes & FILE_ATTRIBUTE_NORMAL) {
+      r["type"] = "Regular";
+    }
+
+    else if (fileInfo.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT) {
+      r["type"] = "Symbolic";
+      r["symlink"] = "1";
+
+    } else {
+      // This is the type returned from GetFileType -> FILE_TYPE_DISK
+      r["type"] = "Disk";
+    }
+    break;
+
+  case FILE_TYPE_PIPE:
+
+    // If GetNamedPipeInfo fails we assume it's a socket
+    (GetNamedPipeInfo(hFile, 0, 0, 0, 0)) ? r["type"] = "Pipe"
+                                          : r["type"] = "Socket";
+    break;
+
+  case FILE_TYPE_UNKNOWN:
+
+    r["type"] = "Unknown";
+    break;
+
+  default:
+    r["type"] = "Unknown";
+  }
+
+  r["attrib"] = getFileAttribStr(fileInfo.dwFileAttributes);
+
+  std::stringstream volumeSerial;
+  volumeSerial << std::hex << std::setfill('0') << std::setw(4)
+               << HIWORD(fileInfo.dwVolumeSerialNumber) << "-" << std::setw(4)
+               << LOWORD(fileInfo.dwVolumeSerialNumber);
+
+  r["device"] = BIGINT(fileInfo.dwVolumeSerialNumber);
+  r["volume_serial"] = TEXT(volumeSerial.str());
+
+  LARGE_INTEGER li = {0};
+  (GetFileSizeEx(hFile, &li) == 0) ? r["size"] = BIGINT(-1)
+                                   : r["size"] = BIGINT(li.QuadPart);
+
+  char* pszDrive = NULL;
+  char szDrive[4];
+  int nDriveLetterIndex = PathGetDriveNumber(path.string().c_str());
+
+  if (nDriveLetterIndex != -1) {
+    if (kDriveLetters.count(nDriveLetterIndex)) {
+      szDrive[0] = kDriveLetters.at(nDriveLetterIndex);
+      szDrive[1] = ':';
```

Comment:
Looks like we always add the `:\\` to the string, maybe let's change `kDriveLetters` to be:
```
const std::map<size_t, std::string> kDriveLetters{
    {0,  "A:\\"}
	{1,  "B:\\"}
	{2,  "C:\\"}
	{3,  "D:\\"}
	{4,  "E:\\"}
	{5,  "F:\\"}
	{6,  "G:\\"}
	{7,  "H:\\"}
	{8,  "I:\\"}
	{9,  "J:\\"}
	{10, "K:\\"}
	{11, "L:\\"}
	{12, "M:\\"}
	{13, "N:\\"}
	{14, "O:\\"}
	{15, "P:\\"}
	{16, "Q:\\"}
	{17, "R:\\"}
	{18, "S:\\"}
	{19, "T:\\"}
	{20, "U:\\"}
	{21, "V:\\"}
	{22, "W:\\"}
	{23, "X:\\"}
	{24, "Y:\\"}
	{25, "Z:\\"}
};
```
And then we don't require the `char` local variables, we can just grab it directly and pass it to `GetDiskFreeSpace`